src/main/java/nl/wearefrank/openapifrankadapter/OpenapiFrankadapterApplication.java

```diff
@@ -56,41 +56,75 @@ public static void main(String[] args) {
     @PostMapping(value = "/receiver-file", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     public ResponseEntity<Resource> postFileReceiver(@RequestParam("file") MultipartFile file) throws IOException, SAXException {
         // Check if it's a JSON file
-        if (!file.getContentType().equals("application/json") && !file.getContentType().equals("application/yaml")) {
+        if (!file.getContentType().matches("application/(json|yaml|x-yaml|octet-stream|text/yaml|plain)")) {
```

Comment:
Als ik het zo goed lees, gaat het laatste stukje niet goed, tussen haakjes match je het stuk na `application/`, maar tussen haakjes staat ook `text/`. Daarnaast vraag ik me af of je hier octet-stream en text/plain wel wilt meenemen? Die zijn nogal generiek.

Reply:
Sterker nog, ik zie 3x dezelfde code ;-) Als ik de hele body van deze method selecteer en 'extract method' in intellij kies, stelt hij zelfs voor om de 'Option' als extra parameter toe te voegen. Daarmee dupliceer je die code niet meer.